### DISCUSSION TITLE
Possibility to plot Line of best fit in a Line Chart?

### DISCUSSION THREAD

**Discussion by klazaj** [apache/superset#18443] [Ideas]
Is there a way to plot Line of best fit (simple linear regression) into a Line chart, with times series data? Does this option exist in Superset, or should it be coded manually into a SQL Lab before applying it to a chart?

**Comment by junlincc** [comment]
good question.... @villebro does annotation support? if not, can we extend this feature to support?

**Comment by villebro** [comment]
Hi, we don't currently have support for this, but it should be fairly straight forward to implement for the new chart types (=the ones using the new chart data API). If someone wants to work on this I can give directions on how to get started.

**Comment by Anugrah-T-Sebastian** [comment]
Hello @villebro. I'm new to Open-source and would love to be part of your community. So can you please give me few directions about where I can get started with this issue? 
I'll be really grateful for your help.

**Comment by villebro** [comment]
@Anugrah-T-Sebastian as a first step it would be really helpful if you could explain, in detail, what the problem you want to solve looks like, and what the requirements and nice-to-haves could look like. For instance, what kind of options would you expect to have to fine tune the estimates etc. After that we can see how that fits into the current codebase and roadmap, after which we can consider how this could be implemented.

**Comment by Anugrah-T-Sebastian** [comment]
@villebro yes. So I was thinking that I can add a feature that takes in an existing file containing 2 attributes (time, time-dependent dimension) schema and generates a linear regression Line graph for it.

The nice-to-haves requirements for me would be:-
- Ability to add different files types containing data
- Option to select the date range
- Simple filters to include or exclude rows in the schema

**Reply by rishirajani-flexmoney** [reply]
Hi @Anugrah-T-Sebastian !
Wanted to know if there was any work done to include these trendlines?
Am highly interested in a Trendline feature for all kinds of charts

**Comment by Cmagno13** [comment]
Ola houve a impltação deste grafico ?

**Comment by miumiups** [comment]
Hi, I'm highly interested in this feature. Would this be implemented? Thank you.

**Comment by polsztyn** [comment]
Wow, I can't believe you don't support a linear regression line.  If you want a specification, I would suggest you just try to replicate the EXCEL Scatter Plot Regression Line functionality.
